include/stronk/unit.hpp

```diff
@@ -78,12 +84,24 @@ struct unit
          * @tparam NewUnitValueT The new unit value type to convert to with same dimensions and underlying type
          * @return a new unit with same dimensions and underlying type, but with the specified scale
          */
-        template<unit_value_like NewUnitValueT>
-            requires(std::same_as<typename NewUnitValueT::unit_t::dimensions_t, dimensions_t>
-                     && std::same_as<UnderlyingT, typename NewUnitValueT::underlying_type>)
-        constexpr auto to() const -> NewUnitValueT
+        template<template<typename UT> typename NewUnitValueT>
```

Comment:
I find this conversion a bit convoluted...
From docstring, it seems that it should only be used to change scales, but I suppose you could also use it to change unit skills (same dimensions, same scale, different skills).

I suppose that is to avoid a unit lookup, if we only took a new scale.

I wonder if this could be made more explicit by not having skills on the units, but only on the values. I think you mentioned last week that it could be done, but I'm unsure what it takes.

Reply:
As discussed we can look into this